### DISCUSSION TITLE
Cinnamon problen at login!

### DISCUSSION THREAD

**Discussion by joellapointe1717** [SpiralLinux/SpiralLinux-project#214] [Q&A]
Hi,
I rebooted my computer. Once at the login screen I did put my password to gain access to my desktop. Then, a popup messages appeared:
"You are currently running in fallback mode.
If you suspect the source of the crash is a local extension, applet or desklet, you can disable downloaded add-ons using the checkbox below.
Do you want to restart Cinnamon?"
I did try the yes and no options with including the checkbox which says: "Disable download applets, desklets and extensions."
Nothing happened, so I did rebooted with the physical button multiple time. I did try to gain access to the terminal with: Ctrl+Alt+T or Fn+T or Windows button+T. My intent was to install an alternative desktop system such as Xfce.

Is there a way to get acess to the terminal?
Furthermore, I'm not a programmation oriented person. I can do some minimal terminal work such as updating, installing software...

Thanks
![20221227_132922](https://user-images.githubusercontent.com/58699551/209747574-0404c4ae-4277-44a8-acee-3316602269ca.jpg)

**Comment by geckolinux** [comment]
Hi there, do you remember what you were doing before the reboot that caused the problem? Were you adding or removing packages or did you update the system?

**Reply by joellapointe1717** [reply]
Sorry, I don't remember what I did before reboot.

**Reply by geckolinux** [reply]
You can check if there are any snapshots in the GRUB boot menu and test them to see if they work.

**Reply by joellapointe1717** [reply]
Yes, there is a big list of snapshots. I did try to boot from the oldest ones. It does the same, no changes.
![20221228_151722](https://user-images.githubusercontent.com/58699551/209868200-be33aad1-6467-4e80-8148-0111ab90382c.jpg)

**Reply by geckolinux** [reply]
OK, that indicates there is something wrong with the settings in your home directory. Did you install any Cinnamon extensions before this happened?

**Reply by joellapointe1717** [reply]
It is possible that I installed some cinnamon extensions to put in the lower bar near the date/time widget. However, I did it a few weeks ago. I was able to boot normally... Is there a way to gain access to the terminal at the login interface or before it?

**Reply by geckolinux** [reply]
Yes, at the Cinnamon failure screen you can hit `Alt` + `Ctrl` + `F2` and login to a terminal with your normal user account. Then try running `mv ~/.config ~/.config-BAK` to temporarily reset your configuration. Then `sudo reboot` and check if that fixes it.

**Reply by joellapointe1717** [reply]
I did the : `mv ~/.config ~/.config-BAK`
It replied : `mv: cannot stat '/root/config': No such file or directory`
Anyway, within the terminal environment, I achieved to install XFCE desktop which is running fine.
However, I would like to make cinnamon to work again. Also I noticed during XFCE install, long rows with this: `dpkg: cannot write to log '/var/log/dpkg.log': Input/output error`

**Reply by joellapointe1717** [reply]
Will reinstall cinnamon do it?

**Reply by geckolinux** [reply]
Hi, please follow this instruction:

> login to a terminal with your normal user account. 

(You logged in as root.)

Reinstalling Cinnamon will not help. It would probably be easiest in your case to reinstall the entire operating system.

**Reply by joellapointe1717** [reply]
Ok, I did `mv ~/.config ~/.config-BAK` with normal user account. It changes nothing...

**Reply by Fantu** [reply]
Hi, have you already checked in ~/.xsession-errors if there are errors?

**Reply by joellapointe1717** [reply]
It does that :
`joel@Spiral:~$ ~/.xsession-errors`
`bash: /home/joel/.xsession-errors: Permission denied`
`joel@Spiral:~$ sudo -s`
`[sudo] password for joel: *******`          
`root@Spiral:/home/joel# ~/.xsession-errors`
`bash: /root/.xsession-errors: No such file or directory`
`root@Spiral:/home/joel# nano ~/.xsession-errors`
`root@Spiral:/home/joel#`

The files doesn't seems to exist!

**Reply by Fantu** [reply]
try `less ~/.xsession-errors` as normal user (when you enter in cinnamon and fail to start) or `less /home/joel/.xsession-errors` for see it (and no execute, is not a binary but text file)
If you do `nano ~/.xsession-errors` will try to edit the root one that normally should not exist
"~" point to the home profile for the current user

**Reply by joellapointe1717** [reply]
I did it and the terminal returned :
`client: GDBus.Error:org.gnome.SessionManager.AlreadyRegistered: Unable to register client
Gtk-Message: 22:25:08.889: Failed to load module "canberra-gtk-module"`

`(caja:1192826): Gtk-WARNING **: 22:25:08.916: Failed to register client: GDBus.Error:org.gnome.SessionManager.AlreadyRegistered: Unable to register client
Gtk-Message: 22:25:09.098: Failed to load module "canberra-gtk-module"`

`(caja:1192831): Gtk-WARNING **: 22:25:09.128: Failed to register client: GDBus.Error:org.gnome.SessionManager.AlreadyRegistered: Unable to register client
Gtk-Message: 22:25:09.311: Failed to load module "canberra-gtk-module"`

`(caja:1192836): Gtk-WARNING **: 22:25:09.335: Failed to register client: GDBus.Error:org.gnome.SessionManager.AlreadyRegistered: Unable to register client
Gtk-Message: 22:25:09.477: Failed to load module "canberra-gtk-module"`

`(caja:1192841): Gtk-WARNING **: 22:25:09.501: Failed to register client: GDBus.Error:org.gnome.SessionManager.AlreadyRegistered: Unable to register client
Gtk-Message: 22:25:09.667: Failed to load module "canberra-gtk-module"`

`(caja:1192846): Gtk-WARNING **: 22:25:09.694: Failed to register client: GDBus.Error:org.gnome.SessionManager.AlreadyRegistered: Unable to register client`
`Gtk-Message: 22:25:09.842: Failed to load module "canberra-gtk-module"
:`

**Reply by Fantu** [reply]
caja is default filemanager of another desktop environment (mate), cinnamon default is nemo instead, so I suppose in latest graphic session you have logged in mate and not cinnamon, right?
for see log related to cinnamon you should try to login in cinnamon and after look .xsession-errors
